actor/builtin/miner/miner.go

```diff
@@ -560,11 +560,20 @@ func (ma *Actor) CommitSector(ctx exec.VMContext, sectorID uint64, commD, commR,
 
 		state.ActiveCollateral = state.ActiveCollateral.Add(collateral)
 
-		if state.Power.Equal(types.NewBytesAmount(0)) {
+		// Case 1: if the miner has never added a sector before start
+		// ther proving period now.  Every miner's first PoSt will be
+		// over a single sector.
+		//
+		// Case 2: if the miner is adding sectors during genesis
+		// construction all committed sectors accumulate in their
+		// proving set.  This is a bit of a hack but it is fairly
+		// clean and allows us to add power immediately in genesis
+		// with commitSector and submitPoSt calls without adding
+		// additional special casing for bootstrappers.
+		if state.SectorCommitments.Size() == 0 || ctx.BlockHeight().Equal(types.NewBlockHeight(0)) {
```

Comment:
This is likely the most controversial thing in this PR.  I was pleased with the simplicity and lack of special casing involved in this solution.  Please take a look and tell me if there is something flawed about it.